core/core-db-repositories/src/main/java/com/sflpro/notifier/db/repositories/repositories/notification/push/PushNotificationRecipientSearchFilter.java

```diff
@@ -0,0 +1,118 @@
+package com.sflpro.notifier.db.repositories.repositories.notification.push;
+
+import com.sflpro.notifier.db.entities.device.mobile.DeviceOperatingSystemType;
+import com.sflpro.notifier.db.entities.notification.push.PushNotificationProviderType;
+import com.sflpro.notifier.db.entities.notification.push.PushNotificationRecipientStatus;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+/**
+ * User: Ruben Dilanyan
+ * Company: SFL LLC
+ * Date: 8/18/15
+ * Time: 1:54 PM
+ */
+public class PushNotificationRecipientSearchFilter {
+
+    /* Properties */
+    private Long subscriptionId;
+
+    private PushNotificationRecipientStatus status;
+
+    private String destinationRouteToken;
+
+    private PushNotificationProviderType providerType;
+
+    private DeviceOperatingSystemType deviceOperatingSystemType;
+
+    private String applicationType;
+
+    /* Constructors */
+    public PushNotificationRecipientSearchFilter() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)